quilt_lite/quilt_lite/lambdas.py

```diff
@@ -0,0 +1,57 @@
+from base64 import b64decode, b64encode
+import importlib
+import sys
+
+from fastapi import FastAPI, HTTPException, Request, Response
+
+sys.path.extend([
+    '../lambdas/shared/',
+    '../lambdas/',
+])
+
+SUPPORTED_LAMBDAS = frozenset([
+    'thumbnail',
+    'preview',
+    's3select',
+    's3proxy',
+])
+
+
+lambdas = FastAPI()
+
+
+@lambdas.api_route("/{name}", methods=['GET', 'POST', 'OPTIONS'])
+@lambdas.api_route("/{name}/{path:path}", methods=['GET', 'POST', 'OPTIONS'])
+async def lambda_request(request: Request, name: str, path: str = ''):
+    if name not in SUPPORTED_LAMBDAS:
+        raise HTTPException(404, "No such lambda")
+
+    req_body = await request.body()
+
+    args = {
+        'httpMethod': request.method,
+        'path': request.url.path,
+        'pathParameters': {
+            'proxy': path
+        },
+        'queryStringParameters': dict(request.query_params) or None,
+        'headers': request.headers or None,  # FastAPI makes headers lower-case, just like AWS.
+        'body': b64encode(req_body),
+        'isBase64Encoded': True
+    }
+
+    module = importlib.import_module(f'{name}.index')
+
+    result = module.lambda_handler(args, None)
```

Comment:
I _just_ realized that FastAPI uses asyncio. So technically, this should run in a thread-pool... Don't know if that matters just yet.

Reply:
i think the only async / non-blocking call in this function is `await request.body`, but the processing is sync / blocking (as far as i understand python async stuff -- it's quite different from node.js in some details) -- we can try using smth like [asyncio.loop.run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) to make lambda handler invocations truly async, but i'm not sure we'll gain anything from it